app/src/main/java/com/example/concordia_campus_guide/Activities/PathsViewModel.java

```diff
@@ -16,40 +16,40 @@ public Place getFrom() {
         return SelectingToFromState.getFrom();
     }
 
-    public Place getEntrance(Place place){
+    public Place getEntrance(Place place) {
         if (place instanceof RoomModel) {
             String campusCode = ((RoomModel) place).getCampus();
             String floorCode = ((RoomModel) place).getFloorCode();
-            String buildingCode = floorCode.substring(0, floorCode.indexOf("-")).toUpperCase();
+            String buildingCode = floorCode.substring(0, floorCode.indexOf('-')).toUpperCase();
             //For indoor direction, no need to return the actual room from the db as the room coordinates will not be used by the PathFinder
             return new RoomModel(place.getCenterCoordinates(), "entrance", buildingCode + "-1", campusCode);
```

Comment:
Why does it need to return roomModel for the entrance of building ? I feel like using WalkingPoint object here is more suitable. What do you think ?

Reply:
It should be a Place